bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/sitequality/TimingWindowDialog.java

```diff
@@ -0,0 +1,285 @@
+package jsky.app.ot.gemini.editor.sitequality;
+
+import edu.gemini.shared.util.HourMinuteFormat;
+import edu.gemini.shared.util.immutable.*;
+import edu.gemini.spModel.gemini.obscomp.SPSiteQuality.TimingWindow;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.FocusAdapter;
+import java.awt.event.FocusEvent;
+import java.awt.event.KeyAdapter;
+import java.awt.event.KeyEvent;
+import java.text.*;
+import java.util.Date;
+import java.util.TimeZone;
+
+class TimingWindowDialog extends JDialog {
+    private static final long MS_PER_MINUTE = 1000 * 60;
+    private static final long MS_PER_HOUR   = MS_PER_MINUTE * 60;
+
+    private static final String INSTRUCTIONS =
+        "Specify a timing window. Starting time should be in the form YYYY-MM-DD hh:mm:ss " +
+        "and will be interpreted in UTC. Window duration and repeat period are specified in " +
+        "hours and minutes.";
+
+    private static final SimpleDateFormat UTC   = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss") {{
```

Comment:
There we go. Let me know if you think this is better, and we can merge this beast, I hope.